### ISSUE TITLE
What's the next public transit tool you want to see in ArcGIS?

### ISSUE THREAD

**Issue by mmorang** [Esri/public-transit-tools#141]
I'm listening.  What's the next public transit tool you want to see in ArcGIS?  What problems are you trying to solve that you'd like to solve more easily?

**Comment by d-wasserman** [comment]
I was going to make an issue for this, but I was going to ask for help or permission to modify the Count High Frequency Routes at Transit tools to help with SB 743 TPA identification. Basically thinking right now, getting route level data with relation to highest quality stop type (rail etc) at each stop. 
[Legal definition: ](https://sfmea.sfplanning.org/CEQA%20Update-SB%20743%20Summary.pdf)
"A “transit priority area” is defined in as an area within one-half mile of an existing or planned major transit stop. A "major transit stop" is defined in Section 21064.3 of the California Public Resources Code as a rail transit station, a ferry terminal served by either a bus or rail transit service, or the intersection of two or more major bus routes with a frequency of service interval of 15 minutes or less during the morning and afternoon peak commute periods. "

Higher level, the most critical priority is an easy scenario editor for GTFS networks. Currently the main options are TBEST/Remix/UrbanAccess/Conveyal for GTFS based analysis generally alongside these GIS tools. TBEST / Conveyal both enable GTFS editing, but it would be great to have scenario oriented GTFS networks in a network data-set so we can integrate multiple modes/other elements available in network data-sets.

**Comment by mmorang** [comment]
With regard to Item 1, this sounds like a good thing to pursue.  I'll send you and e-mail.

Regarding Item 2, I agree.  My team is aware of this need, but there's not really anything I can do about it in the short term.

**Comment by d-wasserman** [comment]
With 1, I hope it is just an edit to the existing Count High Frequency Routes at Stops Tools. I am thinking just a Boolean parameter for "Attach Route Types to Stops". We can discuss that over email. 

Understood.

**Comment by joshuacroff** [comment]
@mmorang thank you for all of your work on tools to analyze gtfs public transit data! I've been using the Count High Frequency Routes at Stops tool which is quite useful but I'm wondering if it would be possible update the tool to include an option to calculate average headways for given periods without setting a threshold? Perhaps even an option to look at multiple time periods? I'm reading what @d-wasserman has commented above and the work that I'm doing is similar to the work they are doing. Let me know what you think!

**Comment by d-wasserman** [comment]
@mmorang I will help out with edits if you are overloaded ATM.

**Comment by mmorang** [comment]
Go for it, @d-wasserman.

**Comment by d-wasserman** [comment]
Trying to get permission (re-permission to share). Basically if I get internally approval, it would be another Fehr & Peers contributed tool. If not, I will propose modifications to the existing count at stops tools. The contributed tool would have have the advantage of being a system "explode". Every stop-route-direction is given a unique geometry that can be used to get route level statistics. The problem with it is you need to remember you get multiple stops (if 3 route-directions pass a stop, it gets 3 stops - each with the frequency information of the routes stopping there). The output is detailed enough to deal with  issues requested here and 145. Basically people could use either spatial joins or dissolves to get the information they want.

**Comment by d-wasserman** [comment]
Another feature request is is it possible to get the transit network solver to identify which routes are used in the course of a trip?
I think now with closest facility or routing layers, this is some what possible by doing multiple join between the various tables in the data model (correct me if I am incorrect?), but it does not look easy. Something that can transform a routing analysis into utilized routes would be helpful.

**Comment by mmorang** [comment]
@d-wasserman No, this is not possible with the current implementation.  It is a limitation of the core solver and network evaluator model itself, not something I can solve.

**Comment by d-wasserman** [comment]
> @d-wasserman No, this is not possible with the current implementation. It is a limitation of the core solver and network evaluator model itself, not something I can solve.

I think with current routing, you could do a spatial join between your routes and the LineVariantElements by the solve. From there you would do two more joins between your Lines & LineVariants tables to get a sense of which routes are used by RouteID. Is that not correct?

**Comment by mmorang** [comment]
Oh, my bad.  You are asking for the GTFS route_id, I believe.  Is that right?  (I thought you were asking for trip_id, which is not possible.)

Spatial join is not the correct approach.  Use the Copy Traversed Source Features tool, which directly returns the individual components of the route, complete with joinable IDs.

Find the components in the Copy Traversed Source Features output corresponding to LineVariantElements, join with the LineVariantElements Table.
LineVariantElements LineVarID-> LineVariants ID
LineVariants LineID -> Lines GRouteID

**Comment by d-wasserman** [comment]
That was what I was thinking, and, Copy Traversed Source Features is a great approach. 

This is very cool.

**Comment by mmorang** [comment]
@joshuacroff Regarding the discussion earlier in this thread: With #148, we have added to BetterBusBuffers the Count Trips at Stops by Route and Direction tool. It breaks out the counts at each stop by route_id and direction_id. Download the newest version of the toolbox here: https://arcg.is/0yr9fr  Does this serve your needs?

**Comment by d-wasserman** [comment]
Yes! This tool should help with this type of summary more expediently. I am glad we could get this to work!

**Comment by joshuacroff** [comment]
@mmorang yes, this is excellent! Thank you for your work on this- this will help us tremendously.
